src/coredump/_UPT_access_fpreg.c

```diff
@@ -28,7 +28,27 @@ int
 _UCD_access_fpreg (unw_addr_space_t as, unw_regnum_t reg, unw_fpreg_t *val,
                    int write, void *arg)
 {
+  struct UCD_info *ui UNUSED = arg;
+
+  if (write)
+    {
+      Debug(0, "write is not supported\n");
+      return -UNW_EINVAL;
+    }
+
+#ifdef __s390x__
+  if (reg < UNW_S390X_F0 || reg > UNW_S390X_F15)
+    goto badreg;
```

Comment:
I think it would be more consistent with the existing codebase to just have the Debug() call and return -UNW_EINVAL here instead of jumping out of the \#ifdef to code that can only be reached from this line.

Reply:
I tried to follow the way this is implemented in _UCD_access_reg() where there is a goto badreg; as error handling for each architecture. I however have to admit that it does not make much sense if only one architecture is handled. I've updated the change and moved the error handling.